### ISSUE TITLE
GRAAL for diplid assemblies?

### ISSUE THREAD

**Issue by dcopetti** [koszullab/GRAAL#15]
Hi,

I wonder if GRAAL will fit my genome project.
I have a plant genome assembly with the following features: 
- estimated genome size: 2.6 Gb, diploid organism, no recent WGD;
- total assembly size: 4.5 Gb, scf N50 3 Mb, scf N80 1 Mb, 1.4% Ns;
- BUSCO genes: 98% present, >70% in two copies
It is indeed a diploid assembly.

I wonder if GRAAL can use allelic variation to produce phased pseudochromosome sequences. 
By collinearity I am able to assign 80% of the sequence to chromosomes (of a closely-related species), but I have pairs of scaffolds at each locus. I would like to split the pairs in the two allelic genomes in a phased fashion. Would GRAAL work with this?
Thanks,

Dario

**Comment by baudrly** [comment]
Hello,

I have indeed been using GRAAL with success to assemble diploid genomes. This depends on how heterozygous the assembly is. If chromosomes are too similar to each other there will be too many mapping issues and 3C-based assemblers are generally unable to distinguish reads mapping onto either member of a chromosome pair. But if the chromosomes can be distinguished I have found that GRAAL is relatively robust to these mapping issues and can separate two chromosomes from a pair, albeit with a noticeable pattern:

![hetero_pattern](https://user-images.githubusercontent.com/16543991/39630282-311797aa-4faf-11e8-82c7-3ed0131c6a22.png)

If that works for you don't hesitate to try it out and report any issue you may have found.

**Comment by dcopetti** [comment]
perfect, that image is what I was looking for! (if the two are allelic chromosomes/scffolds)
do you have any recommendation on how to prepare the Hi-C data? any favorite protocol or method?
Thanks for the feedback,

**Comment by baudrly** [comment]
Yes they are an allelic pair, this is a typical pattern I've found among such chromosomes. As for the generation of GRAAL-compatible contact maps, you may use [HiC-Box](https://github.com/koszullab/HiC-Box) (graphical interface based) or [my own pipeline](https://github.com/baudrly/yahcp) (command-line based). Or you may convert the data manually according to the format described in the [readme](https://github.com/koszullab/GRAAL/blob/master/README.md#dataset-format) if you already have some data at your disposal that's been processed by another Hi-C pipeline. In any case, the pipelines are by and large equivalent for reassembly purposes, so simply pick whatever is the most convenient for you.

Cheers

**Comment by dcopetti** [comment]
great, thanks.
How about the wet lab part, any particular recommendation?

**Comment by baudrly** [comment]
I don't do experiments anymore and I am not familiar on protocol specifics for plants but [here](https://www.ncbi.nlm.nih.gov/pubmed/28623585)'s a very handy and recent reference. I hope that helps.
